src/JNI/com/amazonaws/kinesis/video/producer/jni/Parameters.cpp

```diff
@@ -249,10 +251,194 @@ BOOL setClientInfo(JNIEnv *env, jobject clientInfo, PClientInfo pClientInfo) {
+    
+    methodId = env->GetMethodID(cls, "getKvsRetryStrategyCallbacks", "()Lcom/amazonaws/kinesisvideo/producer/KvsRetryStrategyCallbacks;");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getKvsRetryStrategyCallbacks");
+    } else {
+        kvsRetryStrategyCallbacks = (jobject) env->CallObjectMethod(clientInfo, methodId);
+        CHK_JVM_EXCEPTION(env);
+    }
+    setKvsRetryStrategyCallbacks(env, kvsRetryStrategyCallbacks, &pClientInfo->kvsRetryStrategyCallbacks);
+
 CleanUp:
     return STATUS_FAILED(retStatus) ? FALSE : TRUE;
 }
 
+VOID setKvsRetryStrategy(JNIEnv *env, jobject kvsRetryStrategy, PKvsRetryStrategy pKvsRetryStrategy)
+{
+    jmethodID methodId = NULL;
+    jclass cls = NULL;
+    BOOL nullKvsRetryStrategy = FALSE;
+
+    CHECK(env != NULL && pKvsRetryStrategy != NULL);
+
+    if (kvsRetryStrategy == NULL) {
+        nullKvsRetryStrategy = TRUE;
+        goto CleanUp;
+    }
+
+    cls = env->GetObjectClass(kvsRetryStrategy);
+    if (cls == NULL) {
+        DLOGW("Failed to create Java kvsRetryStrategy class.");
+        nullKvsRetryStrategy = TRUE;
+        goto CleanUp;
+    }
+
+    methodId = env->GetMethodID(cls, "getRetryStrategy", "()Lcom/amazonaws/kinesisvideo/producer/RetryStrategy");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getRetryStrategy, setting pRetryStrategy to null.");
+        pKvsRetryStrategy->pRetryStrategy = NULL;
+    } else {
+        if (env->CallLongMethod(kvsRetryStrategy, methodId) == 0) {
+            pKvsRetryStrategy->pRetryStrategy = NULL;
+        } else {
+            // TODO: Implement once we have support for this in Java, setting to null for safety.
+            pKvsRetryStrategy->pRetryStrategy = NULL;
+        }
+    }
+    
+    methodId = env->GetMethodID(cls, "getRetryStrategyConfig", "()Lcom/amazonaws/kinesisvideo/producer/RetryStrategyConfig");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getRetryStrategyConfig, setting pRetryStrategyConfig to null.");
+        pKvsRetryStrategy->pRetryStrategyConfig = NULL;
+    } else {
+        if (env->CallLongMethod(kvsRetryStrategy, methodId) == 0) {
+            pKvsRetryStrategy->pRetryStrategyConfig = NULL;
+        } else {
+            // TODO: Implement once we have support for this in Java, setting to null for safety.
+            pKvsRetryStrategy->pRetryStrategyConfig = NULL;
+        }
+    }
+
+    methodId = env->GetMethodID(cls, "getRetryStrategyType", "()Lcom/amazonaws/kinesisvideo/producer/RetryStrategyType");
+    if (methodId == NULL) {
+        DLOGW("Couldn't find method id getRetryStrategyType, setting retryStrategyType to EXPONENTIAL_BACKOFF_WAIT.");
+        pKvsRetryStrategy->retryStrategyType = KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT;
+    } else {
+        pKvsRetryStrategy->retryStrategyType = (KVS_RETRY_STRATEGY_TYPE) env->CallIntMethod(kvsRetryStrategy, methodId);
+    }
+
+
+CleanUp:
+    if (nullKvsRetryStrategy) {
```

Comment:
If it is a failure, we do not want to set the values (unless there isnt a cleaner way to do so). The goal is to ensure we do not mask the errors. We need to fail if mandatory conditions are not satisified. In the context of this PR, that would be, having the method IDs set and appropriate members passed wherever relevant.